File: week6/sentimental-readability/readability.py
import string
import re

def main():
    text = input("Text: ")
    words = len(get_words(text))
    L = average(get_letters(text), words)
    S = average(get_sentences(text), words)
    level = round(0.0588 * L - 0.296 * S - 15.8)
    if level < 1:
        print("Before Grade 1")
    elif level > 16:
        print("Grade 16+")
    else:
        print(f"Grade {level}")


def get_letters(text):
    letters = 0
    words = get_words(text)
    for i in words:
        translation_table = str.maketrans('', '', string.punctuation)
        result = i.translate(translation_table)
        letters += len(result)
    return letters


def get_sentences(text):
    dots = text.count(".")
    exclamations = text.count("!")
    interrogations = text.count("?")
    sentences = dots + exclamations + interrogations
    return sentences


def get_words(text):
    words = text.split()
    for i in range(len(words)):
        words[i] = words[i].strip(string.punctuation)
    return words

def average(numbers, words):
    x = (100 * numbers)/words
    return x

main()

File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

void get_words(string s, int *words, int *letters, int *sentences);
double average(int number, int words);

int main(void)
{
    string text = get_string("Text: ");
    int w = 1;
    int l = 0;
    int s = 0;

    get_words(text, &w, &l, &s);

    double L = average(l, w);
    double S = average(s, w);
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

void get_words(string s, int *words, int *letters, int *sentences)
{
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == ' ')
        {
            (*words)++;
        }
        else if (isalpha(s[i]))
        {
            (*letters)++;
        }
        else if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            (*sentences)++;
        }
    }
}

double average(int number, int words)
{
    double x = (double) number / (double) words * 100;
    return x;
}
